nodes/src/nodes/prompt_template/template_engine.py

```
@@ -0,0 +1,290 @@
+# ---------------------------------------------------------------------------
+_BUILTINS = {
+    'now': lambda: datetime.now(timezone.utc).isoformat(),
+    'uuid': lambda: str(_uuid.uuid4()),
+    'random': lambda: str(_random()),
+}
+
+
+# ---------------------------------------------------------------------------
+# Context value resolution
+# ---------------------------------------------------------------------------
+def _resolve(name: str, context: dict):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_resolve`

(ANN202)